### PR TITLE
Auto-inserting items to inventory

### PR DESCRIPTION
**Pull request by cm0x4D**
Sometimes it makes actually sense to auto-install items into nodes (#579) just by right-clicking with the correct item.

This PR tries to generalize the concept and adds this feature to a maximum of possible EA elements.

Ready to merge when

- [x] Replace LampSocket implementation by generic process
- [x] Make the process more flexible (more conditions when to actually insert/replace) 
- Add for more items (ideas welcome)
  - [x] Installing drills into the auto miner
  - [x] Refilling pipes in the auto miner
  - ~~Installing Ferromagnetic Cores~~
  - [x] Placing boosters in machines
  - [x] Upgrading panels with the Solar Tracker
  - [x] Entity filter in Turret 
  - ...

---
This change is [Reviewable](https://reviewable.io/reviews/electrical-age/electricalage/584)

### CODE REVIEW

**Review by AMIDIBOSS — CHANGES_REQUESTED** [review]
Items already in inventory slot may get back into player's inventory

**Comment by cm0x4D** [comment]
> Items already in inventory slot may get back into player's inventory

It is not that simple - what if the itemstack had more than one item? What if the inventory does not have any free slots left? The MC API is not that straightforward either. 

I decided to just drop the existing item and the player can take the item if he wants.

File: src/main/java/mods/eln/transparentnode/autominer/AutoMinerElement.java [hunk 1]
```diff
@@ -30,8 +33,12 @@
 
 public class AutoMinerElement extends TransparentNodeElement  {
 
-	TransparentNodeElementInventory inventory = new TransparentNodeElementInventory(AutoMinerContainer.inventorySize, 64, this);
-	
+	AutoAcceptInventoryProxy accpetingInventory =
```
Comment (Baughn) [inline src/main/java/mods/eln/transparentnode/autominer/AutoMinerElement.java]: This is misspelled. (IDEA conveniently does spell-checking if you ask it do.)

**Comment by cm0x4D** [comment]
ready to merge

**Comment by Baughn** [comment]
Tested on Erisia, and works fine (so far).

**Comment by cm0x4D** [comment]
Ok, everything that makes sense for me is done. This might be already too much, I fear that we might confuse players with some of those...

**Pull request merged**